### ISSUE TITLE
What are the output prediction values for binary classification?

### ISSUE THREAD

**Issue by chialikuo** [dmlc/xgboost#91]
I'm new to using xgboost. I'm trying it on a binary classification problem where the training label is either 0 or 1. Although on https://github.com/tqchen/xgboost/wiki/Binary-Classification, it says "**for binary classification, the output predictions are probability confidence scores in [0,1], corresponds to the probability of the label to be positive.**", I got output prediction values definitely not in [0, 1] on my problem. Are these output values log odds ratio, or other transformations of the probabilities? I just need to make sure on this so I can convert them back to probabilities for my use.

P.S. I'm using xgboost's Python interface.

Thanks.

**Comment by HarryCaveMan** [comment]
I am having a similar issue in R and I do have my objective set to `binary:logistic`. I also had a problem with it not seeing where I set the maximum for early stopping. Still had this problem after I remover both of those. I am using the documentation page you get by typing `?xgboost` in the R console. Below is exactly what I have typed. Note that I checked out the class on `y_train` and it is integer, all of which are 0's and 1's:


   ` fit=xgboost(data=x_train,label=y_train,max.depth = 2,nthread = 4, nround = 1000, early.stop.round=10,objective="binary:logistic",verbose=1,maximize=TRUE)`

**Comment by khotilov** [comment]
A side-issue is that you are asking to *maximize* the default error metric... 

However, I find it hard to believe that the logistic transform (that is used in the default settings `predict` for `objective="binary:logistic"`) could produce values outside of [0,1]. Would you be able to provide a reproducible example, or at least the examples of such predicted values that you see?

**Comment by HarryCaveMan** [comment]
@khotilov, You are right about the [0,1], the outputs are definitely probabilities. I read further into the binary:logistic doccumentation and that is what it's supposed to return I just have to use a threshold as described above. Yeah it didn't make much sense to  me to maximize since GB is minimization problem, but the documentation from cran was not really clear to me. Thanks for the response.

**Comment by khotilov** [comment]
The `maximize=` parameter refers to a specific evaluation metric used and is usually only needed when providing a custom evaluation metric function. It has no effect on the underlying algorithm that minimizes the objective function.

**Comment by mabbs90** [comment]
I have a question that I couldn't justify it for myself. The AUC value that I get from xgboost's iterations is different from the AUC value for the predictions coming from the model fit on the same data. I'd be appreciated if you can explain this to me:

```
require(xgboost)
require(pROC)

## load in the agaricus dataset
data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')
dtrain <- xgb.DMatrix(agaricus.train$data, label = agaricus.train$label, missing = NA)
dtest <- xgb.DMatrix(agaricus.test$data, label = agaricus.test$label, missing = NA)

param <- list(max.depth=2,
              eta=1,
              verbose = TRUE,
              objective="binary:logistic",
              eval_metric = 'auc')
nround = 5

## training the model for two rounds
bst <- xgb.train(param, dtrain, nround, nthread = 2, watchlist = list(train = dtrain))
## [0]	train-auc:0.958228
## [1]	train-auc:0.981413
## [2]	train-auc:0.997070
## [3]	train-auc:0.998757
## [4]	train-auc:0.999298

## predict using first full trees
ypred1 <- predict(bst, dtrain)
ypred1 <- ifelse(ypred1 >= 0.5, 1, 0)

auc(agaricus.train$label, ypred1)
## Area under the curve: 0.9931
```

**Comment by geoHeil** [comment]
I have a similar problem here: http://datascience.stackexchange.com/questions/14527/xgboost-predict-probabilities using binary:logistic I would like to obtain the probabilities. However, the vector I receive from xklearn.predict_proba does not look like probabilities.

**Comment by TVMKISHORE** [comment]
Thank you all for the knowledge

**Comment by johnny5550822** [comment]
@geoHeil What do you mean by "the vector I receive from xklearn.predict_proba does not look like probabilities." Are they larger than 1 or ...? Can you post some of the outputs here?

**Comment by geoHeil** [comment]
@johnny5550822 unfortunately, I can't provide numbers. It just looks like http://datascience.stackexchange.com/questions/14527/xgboost-predict-probabilities?noredirect=1#comment18277_14527 there are only 2 values for the predictions. And that is weird. I would have expected to see several (different ) values

**Comment by johnny5550822** [comment]
@geoHeil I see, because I did not get that graph, my value is pretty make sense.

**Comment by keren42** [comment]
I'm using the python wrapper for xgboost and I'm having trouble understanding the way of computing the probabilities returned by predict_proba(). in random forest for example there are equal weights on each tree and we get the mean of proportions over the relevant leaves in all of the trees. How is it calculated in xgboost? are there different weights? how are they determined?

**Comment by marvin520** [comment]
geoHeil get a 2D array from xgboost sklearn API. i got the same data structure too, and i just used the second value as the probability of label 1
